pkg/reporter/json_reporter.go

```diff
@@ -67,3 +88,206 @@ func (jr JsonReporter) Print(reports []Report) error {
 	fmt.Println(string(jsonBytes))
 	return nil
 }
+
+// Tried to pass the Print function to this function but it didn't work
+// We lose the group output when we do that
+// Possible solution: create functions to handle each grouping
+// and pass the Print function to those functions
+// Similiar to how the group_output.go file is structured
+func (jr JsonReporter) PrintSingleGroup(groupReports map[string][]Report, groupOutput string) error {
+	var report groupReportJSON
+	currentPassed := 0
+	currentFailed := 0
+	totalPassed := 0
+	totalFailed := 0
+	report.Files = make(map[string][]fileStatus)
+	report.Summary = make(map[string][]summary)
+
+	for group, reports := range groupReports {
+		report.Files[group] = make([]fileStatus, 0)
+		report.Summary[group] = make([]summary, 0)
+		currentPassed = 0
+		currentFailed = 0
+		for _, r := range reports {
```

Comment:
This for loop and the next appear to be duplicated in this and the other new print functions. I would prefer to eliminate duplicate code. Please refactor the duplicate code into a separate utility function.